### PR TITLE
Databricks tests

### PR DESCRIPTION
**Pull request by zerodarkzone**
Hi,

I've added tests suport to the databricks backend and while doing it also fixed a couple of problems I've found.

### CODE REVIEW

File: tests/integration/engines/test_int_functions.py [hunk 1]
```diff
@@ -1553,11 +1585,16 @@ def test_timestamp_seconds(get_session_and_func):
...
+    w = (
+        df.withColumn("window", window("date", "5 seconds"))
+        .groupBy("window")
+        .agg(sum("val").alias("sum"))
+    )
     result = w.select(
-        w.window.start.cast("string").alias("start"),
-        w.window.end.cast("string").alias("end"),
+        col("window.start").cast("string").alias("start"),
+        col("window.end").cast("string").alias("end"),
```
Comment (eakmanrq) [inline tests/integration/engines/test_int_functions.py:1597]: Why was this change made? For context I try to stick as close as possible to the examples from PySpark documentation in order to validate consistent behavior/compatibility: https://spark.apache.org/docs/3.4.0/api/python/reference/pyspark.sql/api/pyspark.sql.functions.window.html

File: tests/integration/engines/test_int_functions.py [hunk 2]
```diff
@@ -2911,7 +3036,10 @@ def test_transform_keys(get_session_and_func, get_func):
     upper = get_func("upper", session)
     df = session.createDataFrame([(1, {"foo": -2.0, "bar": 2.0})], ("id", "data"))
     row = df.select(transform_keys("data", lambda k, _: upper(k)).alias("data_upper")).head()
-    assert sorted(row["data_upper"].items()) == [("BAR", 2.0), ("FOO", -2.0)]
+    if isinstance(session, DatabricksSession):
+        assert sorted(row["data_upper"]) == [["BAR", 2.0], ["FOO", -2.0]]
```
Comment (eakmanrq) [inline tests/integration/engines/test_int_functions.py:3040]: Can you provide some context as to why this should be diffferent? It seems like the key difference is that `data_upper` isn't a dict in Databricks. Is that expected? Is there a way to have is conform to the expected data type?

File: tests/integration/engines/test_int_functions.py [hunk 3]
```diff
@@ -3388,7 +3525,7 @@ def test_approx_percentile(get_session_and_func, get_func):
     assert df.select(approx_percentile("value", [0.25, 0.5, 0.75], 1000000)).collect() == [
         Row(value=[0.7264430125286507, 9.98975299938167, 19.335304783039014])
     ]
-    assert df.groupBy("key").agg(approx_percentile("value", 0.5, 1000000)).collect() == [
+    assert sorted(df.groupBy("key").agg(approx_percentile("value", 0.5, 1000000)).collect()) == [
```
Comment (eakmanrq) [inline tests/integration/engines/test_int_functions.py:3528]: Thanks btw for adding `sorted` to these to make them deterministic. 👍

File: sqlframe/databricks/dataframe.py [hunk 4]
```diff
@@ -53,7 +53,7 @@ def _typed_columns(self) -> t.List[CatalogColumn]:
                 CatalogColumn(
                     name=normalize_string(
                         row.col_name, from_dialect="execution", to_dialect="output"
-                    ),
+                    ).replace("`", ""),
```
Comment (eakmanrq) [inline sqlframe/databricks/dataframe.py:56]: Can you provide some context in terms of why this is needed? It looks like `normalize_string` is quoting the values but then you are removing them after.

**Comment by eakmanrq** [comment]
Really appreciate your contribution of these tests. 🙏 

I made a few comments but can you also summarize some of the key differences you are seeing with the Databricks engine compared to the PySpark API? From what I can tell it was the following:
* Timestamp differences (I'm currently refactoring timestamp logic due to a bugfix in SQLGlot that may help. What you have currently is fine though)
* Differences in how arrays/maps are handled

Let me know if there is anything else worth noting. 

In terms of the maps issue, is there a way that we can detect a map and have it return a dict? You can see an example with DuckDB where it checks the value in the response and tries to determine if it is a map and if so converts it to a dict: https://github.com/eakmanrq/sqlframe/blob/ed4c24a52b273460e69087ccdd6a03befc826d5b/sqlframe/duckdb/session.py#L59-L69

File: tests/integration/engines/test_int_functions.py (hunk 1 above)
Comment (zerodarkzone) [reply tests/integration/engines/test_int_functions.py:1597]: Hi,
I've changed it because sqlframe does not support that syntax. Right now this test is being skipped by all the backends except databricks so that´s why it didn't failed before.

Right now the sqlframe Column class does not support the "dot notation" so doing something like `col.start` fails.
image

File: tests/integration/engines/test_int_functions.py (hunk 2 above)
Comment (zerodarkzone) [reply tests/integration/engines/test_int_functions.py:3040]: Hi,
I've found that the databricks sql connector has a couple of "problems/differences" compared with the rest of connectors.

It has different type conversions for some data types.

These are the differences:
The MAP type gets converted to a list of lists instead of a dictionary.
The STRUCT type gets converted to a dictionary instead of a ROW.

I haven't found anything in the connector to change this behaviour but I'll try to look at it a little deeper.

File: sqlframe/databricks/dataframe.py (hunk 4 above)
Comment (zerodarkzone) [reply sqlframe/databricks/dataframe.py]: Hi,
This is not really needed, I've reverted this change

**Comment by zerodarkzone** [comment]
Hi,
I've found a problem with this function.
image
When you have a map, it will covert all the keys to lowercase so this makes this test fails:
image

File: tests/integration/engines/test_int_functions.py (hunk 2 above)
Comment (zerodarkzone) [reply tests/integration/engines/test_int_functions.py]: Hi,

I've kind of solved this but there's a new problem. the `_to_value` method transform all the dict keys to lowercase

**Comment by eakmanrq** [comment]
> When you have a map, it will covert all the keys to lowercase so it makes this test fail:

Yeah I think I know why. Since `upper` is uppercasing the columns in a struct, and they aren't quoted, during normalization it seems the columns as case-insensitive (since they aren't quoted) and therefore the uppercase can become lower case.

This is a tricky case that is more of an edge case. Want to transform the keys in some other ways that can be measured and do an instance check if it is Databricks and do a special test just for Databricks? Maybe add a character or something.

File: tests/integration/engines/test_int_functions.py (hunk 2 above)
Comment (eakmanrq) [reply tests/integration/engines/test_int_functions.py]: See here: https://github.com/eakmanrq/sqlframe/pull/218#issuecomment-2540570341

In short this is some nuance related to normalization and it is fine for now.

File: tests/integration/engines/test_int_functions.py (hunk 1 above)
Comment (zerodarkzone) [reply tests/integration/engines/test_int_functions.py:1591]: Hi, 
I've added a comment on all the tests using this syntax

File: tests/integration/engines/test_int_functions.py (hunk 2 above)
Comment (zerodarkzone) [reply tests/integration/engines/test_int_functions.py]: Hi, I've changed the test for databricks to append a character instead og the upper function.

**Comment by zerodarkzone** [comment]
> > When you have a map, it will covert all the keys to lowercase so it makes this test fail:
> 
> Yeah I think I know why. Since `upper` is uppercasing the columns in a struct, and they aren't quoted, during normalization it seems the columns as case-insensitive (since they aren't quoted) and therefore the uppercase can become lower case.
> 
> This is a tricky case that is more of an edge case. Want to transform the keys in some other ways that can be measured and do an instance check if it is Databricks and do a special test just for Databricks? Maybe add a character or something.

Hi,
This is done

**Review by eakmanrq — APPROVED** [review]
Thanks @zerodarkzone! Feel free to merge if you are ready.

**Comment by zerodarkzone** [comment]
Hi,
I think everything is ready to merge. I don't have permission to do it. Could You merge it?
